connector.rb

```diff
@@ -1,37 +1,2827 @@
+            request = request.payload(payload)
+            request.request_format_www_form_urlencoded
+          end,
+
+          apply: lambda do |connection|
+            if current_url.include?('https://graphql.')
+              access_token = connection['access_token']
+              headers(Authorization: "Bearer #{access_token}") unless access_token.blank?
+            end
+          end,
+
+          refresh_on: [401, 403]
+        }
+      }
     },
 
     base_uri: lambda do |connection|
-      ''
+      instance = connection['instance']
+      region = connection['region']
+      custom_domain_name = connection['custom_domain_name']
+
+      case instance
+      when 'production'
+        case region
+        when 'au'
+          'https://graphql.au.4me.com'
+        when 'uk'
+          'https://graphql.uk.4me.com'
+        when 'us'
+          'https://graphql.us.4me.com'
+        when 'ch'
+          'https://graphql.ch.4me.com'
+        else
+          'https://graphql.4me.com'
+        end
+      when 'quality_assurance'
+        case region
+        when 'au'
+          'https://graphql.au.4me.qa'
+        when 'uk'
+          'https://graphql.uk.4me.qa'
+        when 'us'
+          'https://graphql.us.4me.qa'
+        when 'ch'
+          'https://graphql.ch.4me.qa'
+        else
+          'https://graphql.4me.qa'
+        end
+      when 'demo'
+        'https://graphql.4me-demo.com'
+      else
+        "https://graphql.#{custom_domain_name}"
+      end
     end
   },
 
+  # Tests the connection to the server by performing
+  # a simple introspection query.
   test: lambda do |connection|
+    call(
+      'run_gql',
+      connection,
+      "{
+        __schema {
+          queryType {
+            name
+          }
+        }
+      }",
+      nil,
+      connection['account']
+    )
+  end,
 
+  webhook_keys: lambda do |params, headers, payload|
+    payload['webhook_nodeID']
   end,
 
   actions: {
+    query: {
+      title: 'Query records',
+      subtitle: 'Retrieve one or more records.',
+      help: {
+        body: 'Use this action to get a single record or search all records that matches your search criteria.',
+        learn_more_url: 'https://developer.4me.com/graphql/',
+        learn_more_text: 'Learn more'
+      },
+      description: lambda do |input, picklist_label|
+        name = input['object']
+        name = name[0..-3] if name&.ends_with?('{}')
+        label = name&.labelize&.downcase
+        "Query <span class='provider'>" \
+          "#{label || 'records'}</span> via " \
+          "<span class='provider'>GraphQL</span>"
+      end,
+      input_fields: lambda do |object_definitions|
+        object_definitions['query_input']
+      end,
+      execute: lambda do |connection, input, eis, eos|
+        call('action_execute', connection, input, eis, eos, 'query')
+      end,
+      output_fields: lambda do |object_definitions|
+        object_definitions['query_output']
```

Comment:
@den-sergeyev-workato, this will be a challenge. We have about 420 data entities, so hard-coding them is not an option. I could make it dynamic but that will require a GraphQL query for each field or sub-query selection.  It takes me 20 to 25 click to build my query, this would also mean 20 to 25 request, and we have rate-limiting in place.
I could add the most commonly used data model with a basic sample but that will be a few hundred lines of code.
I will look into it and see what I can do, any suggestions?